### PR TITLE
Markdown Documentation Updates

### LINKED ISSUE #261: Move certain issue descriptions/learnings into project docs
@ian-r-rose will fill in additional detail on this item. 

The ask is to move some of the great background knowledge Ken has shared into project docs. 

This depends on the project docs session on the 18th so putting this in the subsequent sprint.

### PR DESCRIPTION
**Pull request by kengodleskidot**
Markdown files have been added to document the data transformation process, detector health and performance calculations. The index file has also been updated. This fixes #261.

### CODE REVIEW

**Review by thehanggit — COMMENTED** [review]
@kengodleskidot  Everything looks great! Thanks for your comprehensive documentation.

File: docs/data/performance-calcs.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+Lost Productivity: The number of lane-mile-hours on the freeway lost due to operating under congested conditions instead of under free-flow
+conditions. Similar to delay, the user can select different speed threshold values of 35, 40, 45, 50, 55, and 60 MPH. When the freeway
+is in congestion, based on the user selected threshold, the calculation is the ratio between the measured flow and the capacity for this
+location. This drop in capacity is due to the fact that the freeway is operating in congested conditions instead of in free-flow. We
+then multiply one minus this ratio by the length of the segment to determine the number of equivalent lane-miles-hours of freeway which
+this represents. For the capacity we 2076 v/l/h at each location. For example, the formula for Lost Productivity during a 5-minute interval
+for a single station is:
+
+    Lost Productivity = 1 - (5-minute Flow / 173 v/l/5-min) \* Station Length
```
Comment (JamesSLogan) [inline docs/data/performance-calcs.md:48]: The backslash can be removed here: `\*` -> `*`

File: docs/data/performance-calcs.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+is simply the free-flow speed divided by Q:
+
+    TTI = 60 / Q
+
+Lost Productivity: The number of lane-mile-hours on the freeway lost due to operating under congested conditions instead of under free-flow
+conditions. Similar to delay, the user can select different speed threshold values of 35, 40, 45, 50, 55, and 60 MPH. When the freeway
+is in congestion, based on the user selected threshold, the calculation is the ratio between the measured flow and the capacity for this
+location. This drop in capacity is due to the fact that the freeway is operating in congested conditions instead of in free-flow. We
+then multiply one minus this ratio by the length of the segment to determine the number of equivalent lane-miles-hours of freeway which
+this represents. For the capacity we 2076 v/l/h at each location. For example, the formula for Lost Productivity during a 5-minute interval
```
Comment (JamesSLogan) [inline docs/data/performance-calcs.md:45]: What does 2076 mean here?

File: docs/data/detector-health.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
+# Detector Health Diagnostics
+
+Vehicle detectors are the main source of data in the system and provide the best source of real-time traffic data
+available to Caltrans. Data from the detectors is relayed from the field to District TMC's which ultimatley
+makes it's way to the system. The detector data is combined with equipment data
```
Comment (JamesSLogan) [inline docs/data/detector-health.md:5]: `ultimatley makes it's` -> `ultimately makes its`

File: docs/data/detector-health.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+The Detector Health Diagnostic checks performed in the system serve multiple purposes:
+
+1.  Ensure detectors and other equipment are functioning properly
+2.  Serves as a data quality check against data being reported by detectors
+3.  Assists Caltrans staff in troubleshooting and maintaining detectors/stations/controllers
+4.  Used as a data quality measure for imputation, bottleneck and other calculations
+
+Caltrans has installed detectors to cover most urban freeways on the State Highway System. The data collected,
+however, can contain missing values (also referred to as "holes") or “bad” (incorrect) values that require
```
Comment (JamesSLogan) [inline docs/data/detector-health.md:18]: Suggest changing the fancy quotes around `bad` to be straight quotes for consistency

**Review by JamesSLogan — COMMENTED** [review]
Thanks for the considerable documentation effort, Ken! I added a few comments.

One more general thing is that to be a little more user-friendly, it would be great to make links actual markdown links. Aka replace "https://url" with "\[https://url](https://url)" or even "[text describing url]\(https://url)"

File: docs/data/data-transform.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+2. A data pipeline takes the data from the Oracle data lake tables and places it in the Snowflake RAW database. Additional
+   details and a flow diagram are included here: https://cagov.github.io/caldata-mdsa-caltrans-pems/data-loading/
+3. The raw data is then transformed in staging data models and remains aggregated at the station level. The VDS data is also
+   transformed in staging data models.
+4. The 30-second raw data is then aggregated to 5 minute samples per lane in our intermediate data model. For detectors that
+   do not report sample data or where data is missing during a 5-minute interval. The goal of these data models is to capture
+   5-minute intervals for all stations/detectors that are considered active on any given date for all 24 hours.
+5. The 5-minute data aggregations are then processed in the following order:
+
+   - If 10 or more samples are reported in a 5-minute timeframe the sum of the flow value is checked to ensure it is between 0
```
[the remark this answers is not in the document]
Comment (kengodleskidot) [reply docs/data/data-transform.md:47]: The formatting looks good (3 bullets following 5) in dbt and Github on my end. Can you send me a screenshot of what looks off? 
image

File: docs/data/performance-calcs.md (hunk 2 above)
Comment (kengodleskidot) [reply docs/data/performance-calcs.md:45]: The value of 2076 vehicles/lane/hour is the current capacity value used in PeMS for the Productivity performance measure. To maintain consistency, we are using the same value in our Productivity calculation. Here is a screenshot for your reference: 
image

File: docs/data/data-transform.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (JamesSLogan) [reply docs/data/data-transform.md:47]: Ah, I guess we're in a gray area of markdown interpretation if dbt and github get it right. mkdocs generates the following:
image

**Comment by JamesSLogan** [comment]
@kengodleskidot let me push a commit to fix the sub-list since I've tested it out locally

**Review by JamesSLogan — APPROVED** [review]
Thanks Ken!

**Comment by ian-r-rose** [comment]
@kengodleskidot doing some catch up here, and these docs are really wonderful to see. Nice work!

**Pull request merged**